src/plugins/discover/public/application/main/utils/update_saved_search.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { SavedSearch, SortOrder } from '@kbn/saved-search-plugin/public';
+import { DataView } from '@kbn/data-views-plugin/common';
+import { isOfAggregateQueryType } from '@kbn/es-query';
+import { cloneDeep } from 'lodash';
+import { DiscoverAppState } from '../services/discover_app_state_container';
+import { DiscoverServices } from '../../../build_services';
+
+export function updateSavedSearch(
+  {
+    savedSearch,
+    dataView,
+    state,
+    services,
+  }: {
+    savedSearch: SavedSearch;
+    dataView?: DataView;
+    state?: DiscoverAppState;
+    services: DiscoverServices;
+  },
+  updateByFilterAndQuery: boolean = false
```

Comment:
Agreed, you know I'd like to further simplify here , since it's also migrating logic from before, but at a certain point I just had to stop. applied your suggestion

Reply:
Done: https://github.com/elastic/kibana/pull/153528/commits/60b2f525de82c0f887cd0ebcfade99db30ca661a